### ISSUE TITLE
UI - ASP MVC Views, Forms, Identity Additional Clarification

### ISSUE THREAD

**Issue by papyr** [IntentSoftware/Docs#55]
Hello, this is very intriguing hidden gem 💎 , possibly a great approach to building software, if not better replacement to ABP and Serenity, if you can get the word out.

One advantage I see is your UI, is able to support continuous updates and much friendlier

- [ ] How do we scaffold up UI/Views & Controllers with Master Details is there [fluid/templates liquid views](https://github.com/sebastienros/fluid)
- [ ] Is any of the UML or object data passed back from our local machine to Intent Software servers backend when we connect your tool/login. I ask because some of the stuff is sensitive/private and it wasn't clear what was being exposed, my manager may not agree
- [ ] How can we create [dashboard patterns](https://github.com/jdanylko/Tuxboard), with Widgets and portlets/charts apache echarts
- [ ] Backend data abstraction to MongoDb or InlfuxDB (timeseriesDB) Dapper or [YesSql](https://github.com/sebastienros/yessql) 
- [ ] How can we support user driven fields, for e.g. users want to create their own forms, and add fields to it, like [so](https://github.com/optimajet/formbuilder). or like [so](https://github.com/optimajet/DWKit) or [this](https://formbuilder.online/)
- [ ] Would be nice to see how to do also partialviews or viewcomponents, integrate some Bootstrap 5 or Material UI components

**Comment by garethbaars** [comment]
Hi @papyr,

Thanks for reaching out. To answer your questions:

> * [ ]  How do we scaffold up UI/Views & Controllers with Master Details is there [fluid/templates liquid views](https://github.com/sebastienros/fluid)

Automating UI patterns can be done by creating a Module for Intent Architect. Whether this automates a fluid based implementation or angular is dependent on the Module. Modules are created using the Module Builder, which provides an ecosystem for Module creation and maintenance and makes getting set up very quick and easy.

> * [ ]  Is any of the UML or object data passed back from our local machine to Intent Software servers backend when we connect your tool/login. I ask because some of the stuff is sensitive/private and it wasn't clear what was being exposed, my manager may not agree

Nothing that can be considered intellectual property is sent to our servers. Therefore this is no risk that private/sensitive information is exposed to a third party.

> * [ ]  How can we create [dashboard patterns](https://github.com/jdanylko/Tuxboard), with Widgets and portlets/charts apache echarts

Something like this can be achieved through a custom designer and a Module to realize the implementation in code.

> * [ ]  Backend data abstraction to MongoDb or InlfuxDB (timeseriesDB) Dapper or [YesSql](https://github.com/sebastienros/yessql)

Currently we have standard Modules to support EF Core with various providers (e.g. MSSQL, Postgres, Cosmos, etc). We also have a repository pattern to provide a generic abstraction over EF Core. Support for MongoDB, Dapper, etc. would be achieved through a custom Module. It may also be worth mentioning that our team will be building standard Modules for these technologies over time.

> * [ ]  How can we support user driven fields, for e.g. users want to create their own forms, and add fields to it, like [so](https://github.com/optimajet/formbuilder). or like [so](https://github.com/optimajet/DWKit) or [this](https://formbuilder.online/)

While Intent Architect has its own designer system which produces metadata, there is nothing that prevents a developer from providing metadata from other sources (e.g. some of the form builders that you linked). You could then use this metadata to automate the front-end in whichever technology you are using.

> * [ ]  Would be nice to see how to do also partialviews or viewcomponents, integrate some Bootstrap 5 or Material UI components

These features would be achieved through Metadata and Modules.

In summary, Intent Architect is an open platform for pattern reuse. It can be used to automate any text file as long as there is sufficient metadata to describe the developer's "intent". Our current set of standard modules for .NET are focused on backend components and architectures, but there is nothing preventing the tool from being used for frontend development too. Our roadmap in fact will be expanding our Module set into frontend designers (to produce sufficient metadata) and popular technologies (e.g. Angular and React).

If you are interested in using the tool a project, please reach out to us [here](https://intentarchitect.com/#/contact-us) or at [info@intentarchitect.com](mailto:info@intentarchitect.com).

**Comment by papyr** [comment]
Sorry I could not get it to work!
